content: add session formats

cwaring opened the pull request:
ref: protocol/2019-ipfs-camp#30

momack2 commented on src/pages/formats.mdx:
```diff
@@ -0,0 +1,80 @@
+
+## 👩🏽‍🏫 Core & Elective Courses
+
+The Courses of the InterPlanetary Space Training Program (aka IPFS Camp) are one of the main sessions for the event. These are designed to train new IPNauts and the more seasoned IPBuilders in all the arts of the DWeb and P2P.
```
```suggestion
The Core and Elective Courses are the main sessions for the InterPlanetary Space Training Program (aka IPFS Camp). These are designed to train new IPNauts and more seasoned IPBuilders in the arts of the DWeb and P2P.
```

momack2 commented on src/pages/formats.mdx:
```diff
@@ -0,0 +1,80 @@
+
+The Courses of the InterPlanetary Space Training Program (aka IPFS Camp) are one of the main sessions for the event. These are designed to train new IPNauts and the more seasoned IPBuilders in all the arts of the DWeb and P2P.
+
+The Courses are intensive workshops of 75 minutes each. The Core Courses will be a series of sessions to level up their understanding so that we all share a great baseline of understanding, while the Elective Courses will be on even more bleeding edge topics of the DWeb, where each participant decides which ones to attend to advance their knowledge of understanding.
```
```suggestion
The Courses are intensive workshops of 75 minutes each. The Core Courses will be a series of sessions to level-up each participant's understanding so that we all share a great baseline to build upon. The Elective Courses will cover even more bleeding-edge topics of the DWeb, where each participant decides which ones to attend to advance their knowledge and deepen their understanding.
```

momack2 commented on src/pages/formats.mdx:
```diff
@@ -0,0 +1,80 @@
+
+## 📃 Poster Sessions
+
+Poster Sessions are one of the all times IPFS Events favourites. These sessions are designed to propagate knowledge by having small teams of people explore together one of the many pieces of the IPFS stack with the goal to understand it well, so well that they can present it to the whole group aided by a Poster.
```
```suggestion
Poster Sessions are one of the all times favorite IPFS Events. These sessions are designed to propagate knowledge by having small teams of people explore one of the many pieces of the IPFS stack together, with the goal to understand it so well that they can present it to the whole group aided by a Poster.
```

momack2 commented on src/pages/formats.mdx:
```diff
@@ -0,0 +1,80 @@
+
+## ⚡️ Lighting Talks
+
+Lighting Talks are one of the most entertaining and inspiring sections of the IPFS Events. These serve the purpose to enable everyone that is attending the conference to share their own update on their projects with the whole community. Lighting Talks are designed for high throughput, high velocity and large coverage.
```
```suggestion
Lighting Talks are one of the most entertaining and inspiring sections of IPFS Events. These serve to enable everyone attending the conference to share an update on their own projects with the whole community. Lighting Talks are designed for high throughput, high velocity, and large coverage.
```

momack2 commented on src/pages/formats.mdx:
```diff
@@ -0,0 +1,80 @@
+
+# 🐋 Deep Dives
+
+The Deep Dives format is designed to give all participants a taste of what it feels like to work on the big and chunky Open Problems by contributing directly to them in an intense design session, evaluating all the possible solutions, their shortcomings and documenting the results.
```
```suggestion
The Deep Dives format is designed to give all participants a taste of what it feels like to work on the big and meaty Open Problems by contributing directly to them in an intense design session - evaluating all the possible solutions, their advantages and shortcomings, and documenting the results.
```

momack2 commented on src/pages/formats.mdx:
```diff
@@ -0,0 +1,80 @@
+
+The Deep Dives format is designed to give all participants a taste of what it feels like to work on the big and chunky Open Problems by contributing directly to them in an intense design session, evaluating all the possible solutions, their shortcomings and documenting the results.
+
+This type of discussions are core to the genesis of IPFS. We’ve done multiple over the years which yield to many valuable results such as IPLD, PubSub, GraphSync and other advancements to the original IPFS protocol.
```
```suggestion
This type of discussion is core to the genesis of IPFS. We’ve done multiple over the years which have yielded many valuable results such as IPLD, PubSub, GraphSync and other advancements to the original IPFS protocol.
```

momack2 commented on src/pages/formats.mdx:
```diff
@@ -0,0 +1,80 @@
+
+This type of discussions are core to the genesis of IPFS. We’ve done multiple over the years which yield to many valuable results such as IPLD, PubSub, GraphSync and other advancements to the original IPFS protocol.
+
+These design sessions are not about explaining how things work, they were focused on finding the solution with the knowledge and resources we had available.
```
```suggestion
These design sessions are not about explaining how things work, they focus on finding the solution with the knowledge and resources we have available.
```

momack2 commented on src/pages/formats.mdx:
```diff
@@ -0,0 +1,80 @@
+
+## 🧬 Sci-Fi Fair
+
+Do you fancy Science Fairs? Always felt that there was something missing? Something even more futuristic? Despair no more and welcome to the 🧬 IPFS Camp Sci-Fi Fair 🛸. The Sci-Fi fair was created to give a free roaming space for all our attendees to meet all sorts of projects happening from the other attendees of the event.
```
```suggestion
Do you fancy Science Fairs? Always felt that there was something missing? Something even more futuristic? Despair no more and welcome to the 🧬 IPFS Camp Sci-Fi Fair 🛸. The Sci-Fi fair was created to give a free roaming space for all our attendees to meet the awesome projects from other event attendees.
```

cwaring commented:
Thanks for the updates @momack2, I'm going to propagate your changes to https://github.com/ipfs/camp so that we can use those docs as the master for this content. Then (if it's not huge task) I'll aim to source the content directly from the github repo at build-time instead of cloning it here, that way we'll have one source of truth and a simple way to update the copy.

cwaring commented:
ref: protocol/pl-ipfs-team#40 - added location link
ref: protocol/2019-ipfs-camp#30
fixes: #16

cwaring commented:
I've refactored the content input structure on the site and added interactive overlays. Now the schedule is driven by external markdown files located in `content/collections/formats/` with a key for each type.

This is the foundation for when we add the full schedule, which will also be driven by this format.

Main advantages (and upcoming updates):
- Each format/session now has a canonical url `/schedule/formats/{type}` (easy to share - or tweet) 
- Session info can be bi-directionally linked. So when we add the full session descriptions we can link it to the format type so people can navigate to the descriptions with ease and vice versa.
- You are not navigated away from the page, which will be most useful for mobile interactivity before and during the event.
- This pattern is also a reusable component if we wish to adopt elsewhere on the site at a later date. 

![Google Chrome 2019-05-15 at 13 56 56](https://user-images.githubusercontent.com/106938/57777387-b3722200-7719-11e9-9507-9d0ee995909e.png)

I'd like to merge this now and then we can continue to improve the copy and add additional media in another PR.

If there are any changes you'd like to make before merging please LMK.

ref: protocol/2019-ipfs-camp#30

daviddias commented:
> Session info can be bi-directionally linked. So when we add the full session descriptions we can link it to the format type so people can navigate to the descriptions with ease and vice versa.

What is the different between Session Description and Format Description?

daviddias commented:
<img width="789" alt="image" src="https://user-images.githubusercontent.com/1211152/57866511-c3fcc800-77ff-11e9-8f86-9c77dd192f64.png">

cwaring commented:
@daviddias yes agree on core->course, I've committed a change.

Sessions are the events that happen, a format is the type of session.

Here's a draft schema outline to help you understand the relationships:

```ts
type Speaker = {
  name: string
}

type Format = {
  legend: emoji
  type: string
  title: string
  description: string
}

type Session = {
  type: Format
  description: string
  speakers: Speaker[]
}

type Time = {
  startTime: dateString
  endTime: dateString
  session: Session
}

type Day = {
  date: dateString
  title: string
  description: string
  sessions: Time[]
}

type Day = Times[]
type Schedule = Day[]

```

daviddias commented:
Is there a mocked version to see how this looks?
